benchmark/utils/formats.hpp

```diff
@@ -181,6 +181,27 @@ std::unique_ptr<MatrixType> read_matrix_from_data(
     return mat;
 }
 
+
+/**
+ * Creates a CSR strategy of the given type for the given executor if possible,
+ * falls back to csr::classical for executors without support for this strategy.
+ *
+ * @tparam Strategy  one of csr::automatical or csr::load_balance
+ */
+template <typename Strategy>
+std::shared_ptr<csr::strategy_type> create_gpu_strategy(
+    std::shared_ptr<const gko::Executor> exec)
+{
+    if (auto cuda = dynamic_cast<const gko::CudaExecutor *>(exec.get())) {
+        return std::make_shared<Strategy>(cuda->shared_from_this());
```

Comment:
This is a good find! We should definitely document it in `csr.hpp` so that we remember to fix this when we fix #320